### PR TITLE
oidc-generator tests

### PR DESCRIPTION
**Pull request by gugahoi**
happy path tests for oidc generator

### CODE REVIEW

File: konfigurator/oidc_generator_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		httpmock.RegisterResponder(
+			"GET",
+			adfsHostUrl+"/.well-known/openid-configuration",
+			httpmock.NewStringResponder(http.StatusOK, MockResponseOidcConfiguration),
+		)
+
+		konfig, err = NewOidcGenerator(adfsHostUrl, clientId, localPort, localRedirectEndpoint)
+	})
+
+	Context("creating a new OidcGenerator", func() {
```
Comment (pjmichael) [inline konfigurator/oidc_generator_test.go]: would be good to test the error != nil case

File: konfigurator/oidc_generator_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+					"?client_id=%s&redirect_uri=%s&response_type=code&state=%s",
+					clientId,
+					url.QueryEscape("http://localhost:"+localPort+localRedirectEndpoint),
+					state,
+				),
+			)
+		})
+	})
+
+	Context("GetToken", func() {
```
Comment (pjmichael) [inline konfigurator/oidc_generator_test.go]: would be good to test for other scenarios:
- Exchange returns err
- oauth2Token.Extra('id_token') fails

**Comment by prateek-1708** [comment]
Grouse work mate !! LGTM
ship-it

File: konfigurator/oidc_generator_test.go (hunk 2 above)
Comment (gugahoi) [inline konfigurator/oidc_generator_test.go]: description says `happy path tests for oidc generator`

Comment (gugahoi) [inline konfigurator/oidc_generator_test.go]: happy to add tests for the error cases later.

**Comment by prateek-1708** [comment]
👍 + 100 for coveralls ...

**Pull request merged**